### PR TITLE
[Provider] Move to `terraform-plugin-sdk/v2`

### LINKED ISSUE #997: Migrate to `terraform-plugin-sdk` v2
Plugin SDK v2 supports Terraform `0.12.0+`, so I assume it's safe to do already.

For migration details see:
https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
* Update from `terraform-plugin-sdk-v1` to `v2`
* `erros` to `diags`
* Move to context based funcs
* Resolves: #997

## PR Checklist

* [x] Refers to: #997
* [x] Tests added/passed.

## Acceptance Steps Performed

```
=== RUN   TestAccCCENodePoolsV3_basic
--- PASS: TestAccCCENodePoolsV3_basic (1096.06s)
=== RUN   TestAccCCENodePoolsV3_randomAZ
--- PASS: TestAccCCENodePoolsV3_randomAZ (987.89s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

File: opentelekomcloud/provider.go [hunk 1]
```diff
@@ -417,20 +419,14 @@ func Provider() terraform.ResourceProvider {
...
-		return configureProvider(d, terraformVersion)
+	provider.ConfigureContextFunc = func(ctx context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
+		return providerConfigure(ctx, d, provider)
 	}
 
 	return provider
 }
 
-func configureProvider(d *schema.ResourceData, terraformVersion string) (interface{}, error) {
+func providerConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Provider) (interface{}, diag.Diagnostics) {
```
Comment (outcatcher) [inline opentelekomcloud/provider.go]: Context is not used, please make in `_ context.Context`

File: opentelekomcloud/common/importers.go [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 
-import "github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+import (
+	"context"
 
-func ImportAsManaged(d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ImportAsManaged(ctx context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {
```
Comment (outcatcher) [inline opentelekomcloud/common/importers.go]: Should be `func ImportAsManaged(_ context.Context, d *schema.ResourceData, _ interface{}) ([]*schema.ResourceData, error) {`

**Review by outcatcher — CHANGES_REQUESTED** [review]
Please use `_ context.Context` as an argument in functions where context is not used (not all are commented in the review)

File: opentelekomcloud/services/waf/resource_opentelekomcloud_waf_ccattackprotection_rule_v1.go [hunk 3]
```diff
@@ -163,20 +165,20 @@ func resourceWafCcAttackProtectionRuleV1Create(d *schema.ResourceData, meta inte
 	policy_id := d.Get("policy_id").(string)
 	rule, err := ccattackprotection_rules.Create(wafClient, policy_id, createOpts).Extract()
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomcomCloud WAF CC Attack Protection Rule: %s", err)
+		return fmterr.Errorf("Error creating OpenTelekomcomCloud WAF CC Attack Protection Rule: %s", err)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/waf/resource_opentelekomcloud_waf_ccattackprotection_rule_v1.go]: starts with capital letter

File: opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_site_connection_v2.go [hunk 4]
```diff
@@ -145,12 +148,12 @@ func ResourceVpnSiteConnectionV2() *schema.Resource {
...
 }
 
-func resourceVpnSiteConnectionV2Create(d *schema.ResourceData, meta interface{}) error {
+func resourceVpnSiteConnectionV2Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 
 	config := meta.(*cfg.Config)
 	networkingClient, err := config.NetworkingV2Client(config.GetRegion(d))
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomCloud networking client: %s", err)
+		return fmterr.Errorf("Error creating OpenTelekomCloud networking client: %s", err)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/vpn/resource_opentelekomcloud_vpnaas_site_connection_v2.go]: starts with capital letter

File: opentelekomcloud/services/waf/resource_opentelekomcloud_waf_datamasking_rule_v1.go [hunk 5]
```diff
@@ -52,13 +54,13 @@ func ResourceWafDataMaskingRuleV1() *schema.Resource {
...
 
-func resourceWafDataMaskingRuleV1Create(d *schema.ResourceData, meta interface{}) error {
+func resourceWafDataMaskingRuleV1Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*cfg.Config)
 
 	wafClient, err := config.WafV1Client(config.GetRegion(d))
 
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
+		return fmterr.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/waf/resource_opentelekomcloud_waf_datamasking_rule_v1.go]: starts with capital letter

File: opentelekomcloud/services/waf/resource_opentelekomcloud_waf_falsealarmmasking_rule_v1.go [hunk 6]
```diff
@@ -45,13 +47,13 @@ func ResourceWafFalseAlarmMaskingRuleV1() *schema.Resource {
...
 
-func resourceWafFalseAlarmMaskingRuleV1Create(d *schema.ResourceData, meta interface{}) error {
+func resourceWafFalseAlarmMaskingRuleV1Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*cfg.Config)
 
 	wafClient, err := config.WafV1Client(config.GetRegion(d))
 
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
+		return fmterr.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/waf/resource_opentelekomcloud_waf_falsealarmmasking_rule_v1.go]: starts with capital letter

File: opentelekomcloud/services/waf/resource_opentelekomcloud_waf_policy_v1.go [hunk 7]
```diff
@@ -208,13 +210,13 @@ func getAction(d *schema.ResourceData) *policies.Action {
...
 
-func resourceWafPolicyV1Create(d *schema.ResourceData, meta interface{}) error {
+func resourceWafPolicyV1Create(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	config := meta.(*cfg.Config)
 
 	wafClient, err := config.WafV1Client(config.GetRegion(d))
 
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
+		return fmterr.Errorf("Error creating OpenTelekomcomCloud WAF Client: %s", err)
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/waf/resource_opentelekomcloud_waf_policy_v1.go]: starts with capital letter

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]
check last 10 files for error messages

**Review by outcatcher — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**